scene/2d/tile_map.cpp

```diff
@@ -1051,18 +1062,26 @@ void TileMap::draw_tile(RID p_canvas_item, Vector2i p_position, const Ref<TileSe
 
 void TileMap::_physics_notification(int p_what) {
 	switch (p_what) {
+		case CanvasItem::NOTIFICATION_ENTER_TREE: {
+			last_valid_transform = get_global_transform();
+		} break;
+		case CanvasItem::NOTIFICATION_INTERNAL_PHYSICS_PROCESS: {
+			if (sync_to_physics) {
+				last_valid_transform = new_transform;
+				set_notify_local_transform(false);
+				set_global_transform(new_transform);
+				set_notify_local_transform(true);
+			}
+		} break;
 		case CanvasItem::NOTIFICATION_TRANSFORM_CHANGED: {
-			// Update the bodies transforms.
-			if (is_inside_tree()) {
+			if (is_inside_tree() && !sync_to_physics) {
 				for (int layer = 0; layer < (int)layers.size(); layer++) {
-					Transform2D global_transform = get_global_transform();
-
 					for (Map<Vector2i, TileMapQuadrant>::Element *E = layers[layer].quadrant_map.front(); E; E = E->next()) {
 						TileMapQuadrant &q = E->get();
 
 						Transform2D xform;
 						xform.set_origin(map_to_world(E->key() * get_effective_quadrant_size(layer)));
-						xform = global_transform * xform;
+						xform = new_transform * xform;
```

Comment:
It seems this line is causing the issues when sync to physics is disabled, it should use the global transform like before, since in this case `new_transform` is never updated.

Apart from that it seems to all work fine!